docs/examples/api-reference/sources/sql.py

```diff
@@ -218,3 +218,40 @@ class UserClick:
 
     # /docsnip
     client.commit(message="some commit msg", datasets=[UserClick])
+
+
+@mock
+def test_pubsub_basic(client):
+    os.environ["DB_NAME"] = "some-db-name"
+    # docsnip pubsub_source
+    from fennel.connectors import source, PubSub
+    from fennel.datasets import dataset, field
+
+    # docsnip-highlight start
+    pubsub = PubSub(
+        name="pubsub_src",
+        project_id="test_project",
+        service_account_key={
+            "type": "service_account",
+            "project_id": "fake-project-356105",
+            "client_email": "randomstring@fake-project-356105.iam.gserviceaccount.com",
+            "client_id": "103688493243243272951",
+            "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+            "token_uri": "https://oauth2.googleapis.com/token",
+            "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+        },
+    )
+    # docsnip-highlight end
+
+    # docsnip-highlight next-line
+    collection = pubsub.topic("test_topic")
```

Comment:
Rename `collection` to `topic`. In fact, try to inline `pubsub.topic(...)` in the `@source` decorator in a single line - if you can, do that. Else, keep it like this across two lines. You may have to shorten `test_topic` to say `name`